### PR TITLE
Enable nullability in ListViewSubItem

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in ListViewSubItem.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6836)

### CODE REVIEW

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 1]
```diff
@@ -94,9 +94,9 @@ public async Task ListView_Tile_SubItem_FragmentNavigate_Sibling_ReturnsNullAsyn
             {
                 InitializeTileList(listView, columnCount, subItemsCount, tileSize: new Size(100, 100));
 
-                AccessibleObject accessibleObject = listView.Items[0].SubItems[1].AccessibilityObject;
-                IRawElementProviderFragment nextAccessibleObject = accessibleObject.FragmentNavigate(NavigateDirection.NextSibling)!;
-                IRawElementProviderFragment previousAccessibleObject = accessibleObject.FragmentNavigate(NavigateDirection.PreviousSibling)!;
+                AccessibleObject? accessibleObject = listView.Items[0].SubItems[1].AccessibilityObject;
+                IRawElementProviderFragment? nextAccessibleObject = accessibleObject?.FragmentNavigate(NavigateDirection.NextSibling)!;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:98]: you have negation operator telling `accessibleObject` cannot be null at this location.

**Review by dreddy-work — CHANGES_REQUESTED** [review]
Please take a relook at accessibleobject use.

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs (hunk 1 above)
Comment (gpetrou) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs]: I should have removed these.

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewItem.ListViewSubItem.cs [hunk 2]
```diff
@@ -59,8 +58,18 @@ public ListViewSubItem(ListViewItem owner, string text, Color foreColor, Color b
                 };
             }
 
-            internal AccessibleObject AccessibilityObject
-                => _accessibilityObject ??= new ListViewSubItemAccessibleObject(this, _owner);
+            internal AccessibleObject? AccessibilityObject
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListViewItem.ListViewSubItem.cs:61]: @vladimir-krestov/ @Tanya-Solyanik  , does this alter any assumptions or behavior around accessibility object?

**Pull request merged**